qiskit/algorithms/gradients/lin_comb_qfi.py

```diff
@@ -0,0 +1,258 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+A class for the Quantum Fisher Information.
+"""
+
+from __future__ import annotations
+
+from typing import Sequence
+from copy import copy
+
+import numpy as np
+
+from qiskit.algorithms import AlgorithmError
+from qiskit.circuit import Parameter, ParameterExpression, QuantumCircuit
+from qiskit.primitives import BaseEstimator
+from qiskit.providers import Options
+from qiskit.quantum_info import SparsePauliOp
+
+from .base_qfi import BaseQFI
+from .lin_comb_estimator_gradient import DerivativeType, LinCombEstimatorGradient
+from .qfi_result import QFIResult
+from .utils import _make_lin_comb_qfi_circuit
+
+
+class LinCombQFI(BaseQFI):
+    """Computes the Quantum Fisher Information (QFI) given a pure,
+    parameterized quantum state. This method employs a linear
+    combination of unitaries [1].
+
+    **Reference:**
+    [1] Schuld et al., Evaluating analytic gradients on quantum hardware, 2018
+    `arXiv:1811.11184 <https://arxiv.org/pdf/1811.11184.pdf>`_
+
+    """
+
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        phase_fix: bool = True,
+        derivative_type: DerivativeType = DerivativeType.REAL,
+        options: Options | None = None,
+    ):
+        r"""
+        Args:
+            estimator: The estimator used to compute the QFI.
+            phase_fix: Whether to calculate the second term (phase fix) of the QFI, which is
+                Re[(dω⟨<ψ(ω)|)|ψ(ω)><ψ(ω)|(dω|ψ(ω))>]. Default to ``True``.
```

Comment:
Whether it's `Re`, `Im`, or both also depends on the `derivative_type`, right?

Reply:
Right. The documentation was strange. I changed it.